google-cloud-storage/lib/google/cloud/storage/file.rb

```diff
@@ -543,17 +553,32 @@ def update
         #   file.download "path/to/downloaded/gzipped.txt",
         #                 skip_decompress: true
         #
-        def download path = nil, verify: :md5, encryption_key: nil,
+        # @example Partially download.
+        #
+        #   require "google/cloud/storage"
+        #
+        #   storage = Google::Cloud::Storage.new
+        #   bucket = storage.bucket "my-bucket"
+        #   file = bucket.file "path/to/my-file.ext"
+        #
+        #   downloaded = file.download range: "bytes=3-6"
```

Comment:
I like the simplicity of design, but I'm questioning about exposing `range_start:` and `range_end:` parameters instead of having a developer determine the format correctly `bytes=#-#`. 

Another design could be using a Ruby Range `#...#`, so `range: #..#` and benefit from the lazy enumeration.